### DISCUSSION TITLE
How to hide nuclei templates list from getting displayed on terminal?

### DISCUSSION THREAD

**Discussion by njmulsqb** [projectdiscovery/nuclei#446] [Q&A] [answered]
Hello,

I am using your amazing tool and I have one query, whenever I run nuclei it displays all the templates that I am using in my case they're 300+, is there any way I can stop nuclei from doing this? I just want to see results on the terminal. I checked the help page of nuclei but couldn't find something related to this. 

Thanks

**Comment by geeknik** [comment, marked as answer]
Try the `-silent` command line flag. 👍🏻

**Reply by ehsandeep** [reply]
@najum98 as geeknik suggested, `silent` flag should only display results.

**Reply by njmulsqb** [reply]
Thank you @geeknik and @team-projectdiscovery

**Comment by kinthaiofficial** [comment]
To suppress the template list output in nuclei, a few approaches depending on what you're seeing:

**1. Suppress template loading output**:
```bash
# -silent flag suppresses most output
nuclei -u https://target.com -t templates/ -silent

# Or use -no-color to get cleaner output without color codes
nuclei -u https://target.com -t templates/ -no-color
```

**2. Filter output to results only**:
```bash
# Only show findings, not progress/template info
nuclei -u https://target.com -t templates/ -silent -o results.txt

# Or with jsonl output for processing
nuclei -u https://target.com -t templates/ -json -o results.jsonl 2>/dev/null
```

**3. Disable specific output types**:
```bash
# Suppress stats
nuclei -u https://target.com -disable-update-check -stats=false

# In config file (~/.config/nuclei/config.yaml)
disable-update-check: true
stats: false
silent: true
```

**4. Redirect stderr** (template list usually goes to stderr):
```bash
nuclei -u https://target.com -t templates/ 2>/dev/null
```

**5. Use `-T` flag to specify templates quietly**:
```bash
# Point directly to specific template instead of directory
nuclei -u https://target.com -T cves/2023/CVE-2023-XXXXX.yaml
```

For CI/CD pipelines where you want clean output, the combination of `-silent -json -o results.jsonl 2>/dev/null` gives you just the JSON findings with no noise.
